src/utils/utilities.jl

```diff
@@ -62,12 +62,14 @@ individual velocity components:
 function compute_kinetic!(κ::Fields.Field, uₕ::Fields.Field, uᵥ::Fields.Field)
     @assert eltype(uₕ) <: Union{C1, C2, C12}
     @assert eltype(uᵥ) <: C3
-    @. κ =
-        1 / 2 * (
-            dot(C123(uₕ), CT123(uₕ)) +
-            ᶜinterp(dot(C123(uᵥ), CT123(uᵥ))) +
-            2 * dot(CT123(uₕ), ᶜinterp(C123(uᵥ)))
-        )
+    NVTX.@range "compute_kinetic! kernel" color = colorant"brown" begin
```

Comment:
why's the word kernel used? is it a kernel invocation here (and not the other places?)

Reply:
In this case, we are just profiling the cuda kernel inside the `compute_kinetic!` function instead of the entire function. The `.` broadcasting syntax automatically launches a CUDA kernel on a GPU!

Reply:
I see. Thanks!